### PR TITLE
fix: [M3-8872] - Table and Chart Legend Improvements

### PR DESCRIPTION
**Pull request by jaalah-akamai**
## Description 📝
We needed to reduce the spacing for our table headers to match our CDS as well as within our legend table.

## Changes  🔄
- Added a space between legend tooltip value and unit
- Reduced the height of the table header across the board from 46 to 40 to match guidance from CDS
- Removed additional padding for legends
- Added ability to use built-in y-axis formatter

## Target release date 🗓️
12/10

## Preview 📷

| Before  | After   |
| ------- | ------- |
| Screenshot 2024-11-20 at 10 06 47 AM | Screenshot 2024-11-20 at 10 06 37 AM |
| localhost_3000_linodes_67142831 (3) | localhost_3000_linodes_67142831 |
| localhost_3000_linodes_67142831 (2) | localhost_3000_linodes_67142831 (1) |

### Reproduction steps
- Observe style changes via prod and this branch 
- Linodes landing table
- Linodes detail page has the network charts

### Verification steps
- Verify header changes look good across the board

## As an Author, I have considered 🤔

- 👀 Doing a self review
- ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- 🤏 Splitting feature into small PRs
- ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- 🧪 Providing/improving test coverage
- 🔐 Removing all sensitive information from the code and PR description
- 🚩 Using a feature flag to protect the release
- 👣 Providing comprehensive reproduction steps
- 📑 Providing or updating our documentation
- 🕛 Scheduling a pair reviewing session
- 📱 Providing mobile support
- ♿  Providing accessibility support

<br/>

- [x] I have read and considered all applicable items listed above.

## As an Author, before moving this PR from Draft to Open, I confirmed ✅

- [x] All unit tests are passing
- [x] TypeScript compilation succeeded without errors
- [x] Code passes all linting rules

### CODE REVIEW

**Comment by jaalah-akamai** [comment]
Will branch soon...

File: packages/manager/src/features/CloudPulse/shared/CloudPulseResourcesSelect.tsx [hunk 1]
```diff
@@ -166,13 +169,45 @@ export const CloudPulseResourcesSelect = React.memo(
         }}
         textFieldProps={{
           InputProps: {
-            sx: {
+            sx: (theme) => ({
+              '::-webkit-scrollbar': {
```
Comment (venkymano-akamai) [inline packages/manager/src/features/CloudPulse/shared/CloudPulseResourcesSelect.tsx:173]: @jaalah-akamai , thanks for making scroll bar invisible, but this webkit-scrollbar works for chrome and safari, whereas for firefox and IE/Edge, we need to add following 

```
msOverflowStyle: 'none',
scrollbarWidth: 'none', 
```

File: packages/manager/src/features/CloudPulse/shared/CloudPulseResourcesSelect.tsx [hunk 2]
```diff
@@ -166,13 +169,45 @@ export const CloudPulseResourcesSelect = React.memo(
...
+                  transparent
+                ),
+
+                /* Bottom Cover */
+                linear-gradient(
+                  transparent,
+                  ${theme.tokens.interaction.Background.Primary} 70%
+                ) 0 100%,
+
+                /* Top shadow */
```
Comment (kmuddapo) [inline packages/manager/src/features/CloudPulse/shared/CloudPulseResourcesSelect.tsx:189]: @jaalah  Does this shadows been approved by UX already?  @gitkcosby

File: packages/manager/src/features/CloudPulse/shared/CloudPulseResourcesSelect.tsx (hunk 1 above)
Comment (jaalah-akamai) [reply packages/manager/src/features/CloudPulse/shared/CloudPulseResourcesSelect.tsx:173]: Added

File: packages/manager/src/features/CloudPulse/shared/CloudPulseResourcesSelect.tsx (hunk 2 above)
Comment (jaalah-akamai) [reply packages/manager/src/features/CloudPulse/shared/CloudPulseResourcesSelect.tsx]: Removed, UX will figure out another approach

**Review by nikhagra-akamai — APPROVED** [review]

File: packages/manager/src/components/AreaChart/utils.ts [hunk 3]
```diff
@@ -16,7 +16,7 @@ export const tooltipLabelFormatter = (timestamp: number, timezone: string) =>
   );
 
 export const tooltipValueFormatter = (value: number, unit: string) =>
-  `${roundTo(value)}${unit}`;
+  `${roundTo(value)} ${unit}`;
```
Comment (mjac0bs) [inline packages/manager/src/components/AreaChart/utils.ts:19]: Any reason why we're not doing the same in the legend itself?

Screenshot 2024-11-25 at 2 22 34 PM

**Review by mjac0bs — APPROVED** [review]
Confirmed the spacing is improved and expected test coverage passes.

https://github.com/user-attachments/assets/19f6b14e-ba16-4da0-97f9-32f21890d8c8

**Comment by jaalah-akamai** [comment]
I added the ability to use the built-in formatter, so you can apply whatever you need in that case.

**Review by coliu-akamai — APPROVED** [review]
✅ styling changes in tables in graphs + across the board

thank you!

**Review by jdamore-linode — APPROVED** [review]

**Pull request merged**